src/driver.ml

```diff
@@ -496,6 +496,36 @@ let get_whole_ast_passes ~hook ~expect_mismatch_handler ~tool_name ~input_name =
   linters @ preprocess @ make_generic before_instrs @ make_generic cts
   @ make_generic after_instrs
 
+exception ExceptionFromPPX of (string * exn)
+
+let () =
+  Printexc.register_printer (fun exn ->
+      match exn with
+      | ExceptionFromPPX (owner_name, exn') ->
+          Some
+            (match owner_name with
+            | "<builtin:context-free>" ->
+                Printf.sprintf
+                  {|The following exception was raised during the execution of a context free rewriter:
+%s
+The exception does not contain any information about the location of the error. If you know the responsible ppx, you might want to file an issue to the ppx authors.|}
+                  (Printexc.to_string exn')
+            | _ ->
+                Printf.sprintf
+                  {|The following exception was raised during the execution of the ppx "%s":
+%s
+The exception does not contain any information about the location of the error. You might want to file an issue to the ppx authors.|}
+                  owner_name (Printexc.to_string exn'))
+      | _ -> None)
+
+let process_exn exn owner_name =
+  match Location.Error.of_exn exn with
+  | None -> ExceptionFromPPX (owner_name, exn)
+  | Some error ->
+      Location.Error
+        (Location.Error.set_message error
+           ("(ppx " ^ owner_name ^ ") " ^ Location.Error.message error))
```

Comment:
I went for `(context-free ppx pass) ` as an exception can occur without being related to a specific ppx. So the only information we can give is that it happened during the phase of the driver where it applies (in one pass) the context-free ppxes. When #291 is done, this will precisely be exceptions not thrown by a specific ppx but by the driver.